sdk/core/azure-core/azure/core/pipeline/policies/base.py

```diff
@@ -41,62 +41,101 @@
 
     @abc.abstractmethod
     def send(self, request):
         # type: (PipelineRequest) -> PipelineResponse
-        """Mutate the request.
+        """Abstract send method for a synchronous pipeline. Mutates the request.
+
         Context content is dependent on the HttpTransport.
+
+        :param request: The pipeline request object
+        :type request: ~azure.core.pipeline.PipelineRequest
+        :return: The pipeline response object.
+        :rtype: ~azure.core.pipeline.PipelineResponse
         """
 
 class SansIOHTTPPolicy(Generic[HTTPRequestType, HTTPResponseType]):
     """Represents a sans I/O policy.
-    This policy can act before the I/O, and after the I/O.
-    Use this policy if the actual I/O in the middle is an implementation
-    detail.
-    Context is not available, since it's implementation dependent.
-    if a policy needs a context of the Sender, it can't be universal.
-    Example: setting a UserAgent does not need to be tied to
-    sync or async implementation or specific HTTP lib
+
+    SansIOHTTPPolicy is a base class for policies that only modify or
+    mutate a request based on the HTTP specification, and do no depend
+    on the specifics of any particular transport. SansIOHTTPPolicy
+    subclasses will function in either a Pipeline or an AsyncPipeline,
+    and can act either before the request is done, or after.
     """
 
     def on_request(self, request, **kwargs):
         # type: (PipelineRequest, Any) -> None
-        """Is executed before sending the request to next policy.
+        """Is executed before sending the request from next policy.
+
+        :param request: Request to be modified before sent from next policy.
+        :type request: ~azure.core.pipeline.PipelineRequest
         """
 
     def on_response(self, request, response, **kwargs):
         # type: (PipelineRequest, PipelineResponse, Any) -> None
         """Is executed after the request comes back from the policy.
+
+        :param request: Request to be modified after returning from the policy.
+        :type request: ~azure.core.pipeline.PipelineRequest
+        :param response: Pipeline response object
+        :type response: ~azure.core.pipeline.PipelineResponse
         """
 
     #pylint: disable=no-self-use
     def on_exception(self, _request, **kwargs):  #pylint: disable=unused-argument
         # type: (PipelineRequest, Any) -> bool
-        """Is executed if an exception comes back from the following
-        policy.
-        Return True if the exception has been handled and should not
-        be forwarded to the caller.
+        """Is executed if an exception is raised while executing the next policy.
+
+        Developer can optionally implement this method to return True
+        if the exception has been handled and should not be forwarded to the caller.
+
         This method is executed inside the exception handler.
-        To get the exception, raise and catch it:
+
+        .. code-block:: python
```

Comment:
do we want to move this as well to the test_examples file with a more concrete impl?